internal/execute/tableutil/builder_cache.go

```diff
@@ -0,0 +1,119 @@
+package tableutil
+
+import (
+	"reflect"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+)
+
+// Builder is the minimum interface for constructing a Table.
+type Builder interface {
```

Comment:
And this is only used by `ArrowBuilder` and `BufferedTableBuilder` right?

Reply:
You are correct that it's only used by those two structs. This isn't intended to be related to `execute.TableBuilder`. This is intended as an experiment for what it should be instead.

The intention behind calling it `Builder` is because the package is named `tableutil`. So it will get referenced as `tableutil.Builder` in packages outside of this. Effective Go says that APIs should rely on this to avoid stutter in naming: https://golang.org/doc/effective_go.html#names

I had been thinking that `tableutil` might be a bad package name. Do you think renaming the package to `table` would alleviate some of these issues? That way, it gets referenced as `table.Builder`.

Reply:
Yeah `table.Builder` sounds good to me.